### PR TITLE
[IMP]Website Blog: Make Onboarding tour user friendly and change customi...

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
...ze cover feature
- Task: https://www.odoo.com/web?debug=#id=8996&view_type=form&model=project.task&menu_id=3942&action=327
- Pad: https://pad.odoo.com/p/r.MnhQk73bZbMtu2Mv

### CODE REVIEW

File: addons/website_blog/static/src/js/website_blog.editor.js [hunk 1]
```diff
@@ -76,9 +90,34 @@
...
             editor.on('saved', self, function (event, img) {
                 var url = self.$image.attr('src');
-                self.$target.css({"background-image": url ? 'url(' + url + ')' : "", 'min-height': $(window).height()});
+                self.$target.find('#js_blogcover').css({"background-image": url ? 'url(' + url + ')' : ""});
+                self.$target.attr("class","cover cover_full");
+                this.$el.parent().find('.snippet-option-website_blog:not(li[data-change])').removeClass("hidden");
             });
         },
+        cover_class : function(type, value, $li) {
+            this.$target.attr("class", type === 'over' || 'click' ? value : this.class);
```
Comment (vme-odoo) [inline addons/website_blog/static/src/js/website_blog.editor.js]: above condition will return *value* in any condition.

File: addons/website_blog/static/src/js/website_blog.editor.js [hunk 2]
```diff
@@ -76,9 +90,34 @@
...
+        },
+        opacity : function(type, value, $li) {
+            this.$target.find('#js_blogcover').css("opacity", type === 'over' || 'click' ? value : this.value);
```
Comment (vme-odoo) [inline addons/website_blog/static/src/js/website_blog.editor.js]: Same as above

File: addons/website_blog/static/src/js/website_blog.editor.js [hunk 3]
```diff
@@ -76,9 +90,34 @@
...
+        },
+        bgcolor : function(type, value, $li) {
+            this.$target.find('#js_blogcover').attr("class", type === 'over' || 'click' ? value : this.background);
```
Comment (vme-odoo) [inline addons/website_blog/static/src/js/website_blog.editor.js]: same as above

File: addons/website_links/static/src/js/website_links_menu.js [hunk 4]
```diff
@@ -4,9 +4,25 @@
...
+                        waitNot:   '.modal:has(#mobile-viewport) button[data-dismiss=modal]',
+                        element:   '#promote-menu-button',
+                        placement: 'left',
+                        title:     _t("Promote"),
+                        content:   _t("Promote your page by using our SEO optimizing tool or our analytic tracker."),
+                        popover:   { fixed: true },
+                    });
+                }
+            });
+        }
```
Comment (vme-odoo) [inline addons/website_links/static/src/js/website_links_menu.js]: Why blog step mention in *website_links_menu.js* ???

File: addons/document/document.py [hunk 5]
```diff
@@ -97,8 +97,8 @@ def search(self, cr, uid, args, offset=0, limit=None, order=None, context=None,
         # Filter out documents that are in directories that the user is not allowed to read.
         # Must use pure SQL to avoid access rules exceptions (we want to remove the records,
         # not fail), and the records have been filtered in parent's search() anyway.
-        cr.execute('SELECT id, parent_id from ir_attachment WHERE id in %s', (tuple(ids),))
-
+        query_str = """SELECT id, parent_id from ir_attachment WHERE id in %%s ORDER BY %s""" % (self._order if not order else order)
+        cr.execute(query_str, (tuple(ids), ))
         # cont a dict of parent -> attach
```
Comment (jem-odoo) [inline addons/document/document.py]: I think you can remove those change. The order of images in the modal was already fixed, but test it to be sure ;)

File: addons/website/static/src/js/website.contentMenu.js [hunk 6]
```diff
@@ -12,7 +12,9 @@
             self.$el.on('click', 'a[data-action]', function(ev) {
                 ev.preventDefault();
                 var $content_item = $(this);
-                self[$content_item.data('action')]();
+                var data_action = self[$content_item.data('action')];
+                if (data_action)
+                    data_action();
```
Comment (jem-odoo) [inline addons/website/static/src/js/website.contentMenu.js]: Is this change still required ? If averything works without it, remove it.
If you keep it, add '{}' toavoid error when minified.

File: addons/website_blog/static/src/css/website_blog.css [hunk 7]
```diff
@@ -164,10 +172,39 @@ div#blog_angle_down a:hover {
...
+
+.o_none {
+  background-color: #FFFFFF;
+}
+.o_yellow {
+  background-color: #A2A51B;
+}
+.o_green {
+  background-color: #149F2C;
+}
```
Comment (jem-odoo) [inline addons/website_blog/static/src/css/website_blog.css]: move those 3 colors to website.css (website module), around L770 with the others colors. Change the prefix to 'oe_'

File: addons/website_blog/views/website_blog_templates.xml [hunk 8]
```diff
@@ -133,36 +133,34 @@
...
-                    <a t-attf-href="/blog/#{ slug(blog_post.blog_id) }/post/#{ slug(blog_post) }" t-field="blog_post.name" class="mb4"/>
-                  </h2>
-
-                  <div class="text-muted">
-                      <h4
-                        t-field="blog_post.subtitle"
-                        class="mb4 mt4"/>
-                    <div name='blog_post_data' class='mb0'>
-                         <span t-field="blog_post.author_id" style="display: inline-block;" t-field-options='{
+                    <t t-if="has_author_avatar(blog_post.author_id.id)">
```
Comment (jem-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: Test on `if blog_post.author_avatar:`, since this field will be false if no image is defined.

File: addons/website_blog/views/website_blog_templates.xml [hunk 9]
```diff
@@ -187,49 +185,50 @@
...
             </div>
-            <div t-if="blog_post.background_image" id="blog_angle_down">
-                <strong><a href="#blog_content" class="fa fa-angle-down fa-3x fa-inverse mt32"/></strong>
+            <div class="blog_title">
+                <h1 t-field="blog_post.name" id="blog_post_name" t-att-data-blog-id="blog_post.id" placeholder="Blog Post Title"/>
+                <h2 t-field="blog_post.subtitle" placeholder="Subtitle"/>
+                <p class="post-meta text-muted text-center" name="blog_post_data"/>
+                <div>
+                    <t t-if="has_author_avatar(blog_post.author_id.id)">
+                        <span t-field="blog_post.author_avatar" class="o_not_editable" t-field-options='{"widget": "image", "class": "img-circle", "style":"width: 30px; margin-right: 10px; display:inline"}' />
```
Comment (jem-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: same remark as above

File: addons/website/static/src/js/website.contentMenu.js (hunk 6 above)
Comment (htr-odoo) [inline addons/website/static/src/js/website.contentMenu.js]: Yes , still required.
i will add '{}' this.

File: addons/website_blog/static/src/css/website_blog.css (hunk 7 above)
Comment (htr-odoo) [inline addons/website_blog/static/src/css/website_blog.css]: ok.

File: openerp/addons/base/res/res_partner_view.xml [hunk 10]
```diff
@@ -206,7 +206,7 @@
                                     <field name="mobile"/>
                                     <field name="fax"/>
                                     <field name="state_id"/>
-                                    <field name="has_image"/>
+                                    <field name="image"/>
```
Comment (jem-odoo) [inline openerp/addons/base/res/res_partner_view.xml]: Can you put the changes relative to base/res in a separated commit ? Thanks ;)

File: addons/website_blog/controllers/main.py [hunk 11]
```diff
@@ -62,6 +63,9 @@ def nav_list(self, blog=None):
             group['date_end'] = '%s' % datetime.date.strftime(end_date, tools.DEFAULT_SERVER_DATE_FORMAT)
         return groups
 
+    def has_author_avatar(self, author_id):
```
Comment (jem-odoo) [inline addons/website_blog/controllers/main.py]: remove this method, and don't give it in values to render qweb template. In template, test if the related field `avatar_author` is false.

File: addons/website_crm_score/static/src/js/set_score.js [hunk 12]
```diff
@@ -25,10 +25,10 @@
         },
         is_tracked: function(val) {
             var obj = website.seo.Configurator.prototype.getMainObject();
-            if (!obj) {
-                return $.Deferred().reject();
-            } else {
+            if (obj && obj.model == 'ir.ui.view') {
```
Comment (jem-odoo) [inline addons/website_crm_score/static/src/js/set_score.js]: can you enlight me about those changes in set_score.js ?

Comment (htr-odoo) [inline addons/website_crm_score/static/src/js/set_score.js]: @jem-odoo 
In website_crm_score, one field add *track* in ir.ui.view, but in set_score.js when we promote specific blog its takes *blog.post* not ir.ui.view. Same thing can happen with any other page.
Because of this runbot shows warning. 
So did this change.
Thanks.

File: openerp/addons/base/res/res_partner_view.xml (hunk 10 above)
Comment (htr-odoo) [inline openerp/addons/base/res/res_partner_view.xml]: https://github.com/odoo-dev/odoo/compare/odoo:master...master-baserespartner-removefield-htr

File: addons/document/document.py (hunk 5 above)
Comment (htr-odoo) [inline addons/document/document.py]: I tested it , now no need to use it. so i remove this change.

File: addons/website_blog/views/website_blog_templates.xml (hunk 8 above)
Comment (htr-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: done.

File: addons/website_blog/controllers/main.py (hunk 11 above)
Comment (htr-odoo) [inline addons/website_blog/controllers/main.py]: done.

**Comment by jem-odoo** [comment]
Merged in master at f2987ea4126e7b7c27121bccf2aab3a60b229eb3
